src/pybind/mgr/cephadm/module.py

```diff
@@ -2438,6 +2461,11 @@ def _apply(self, spec: GenericSpec) -> str:
 
         return self._apply_service_spec(cast(ServiceSpec, spec))
 
+    @handle_orch_error
+    def service_discovery_dump_cert(self) -> str:
+        root_cert = self.get_store('service_discovery/root/cert')
+        return root_cert if root_cert else 'No cert'
```

Comment:
```suggestion
        return root_cert if root_cert else 'No certificate found for service discovery'
```
Any maybe some info on what the next action should be (e.g. wait a minute and run again) if you have any idea when we're likely to not find the cert.